Create rule S6452 (#1244)
SonarSource/rspec: rules/S6452/java/rule.adoc

> Javadoc code snippets behaves like text-blocks in terms of padding. It is the position of the ending curly brace which defines how many spaces should be placed before each line of the code snippet. Consequently, when writing code snippets, it is recommended to align the curly brace closing the snippet with the first character of the previous lines. This will minimize the number of spaces using in front of each line in the generated javadoc.

Reviewer: ```suggestion
Javadoc code snippets behave like text-blocks in terms of padding. It is the position of the ending curly brace which defines how many spaces should be placed before each line of the code snippet. Consequently, when writing code snippets, it is recommended to align the curly brace closing the snippet with the leftmost character present in the code. This will minimize the number of spaces used in front of each line in the generated javadoc.
```

== Noncompliant Code Example

[source,java]
----
/**
 * {@snippet :
 *   class HelloWorld { ... }

>  * } // Noncompliant - aligned with the snippet's closing bracket, but will force 2 spaces before each lines of the snippet

Reviewer: ```suggestion
 * } // Noncompliant - aligned with the snippet's closing bracket, but will force 2 spaces before each line of the snippet
```

 */
void wrongIndentation();
----

== Compliant Solution

[source,java]
----
/**

>  * Highlighting based on a substring on a single line.

Reviewer: ```suggestion
```

 * {@snippet :
 *   class HelloWorld { ... }

>  *   } // Compliant - aligned with the last curly braces and the class definition starting column of the code above

Reviewer: ```suggestion
 *   } // Compliant - aligned with the leftmost character present in the code, i.e., the 'c' of class.
```

 */
void correctIndentation();
----

== See

> Oracle’s Programmer's Guide to Snippets: https://docs.oracle.com/en/java/javase/18/code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]

Reviewer: ```suggestion
Oracle’s Programmer's Guide to Snippets: https://docs.oracle.com/.../code-snippet/index.html[https://docs.oracle.com/en/java/javase/18/code-snippet/index.html]
```

ifdef::env-github,rspecator-view[]

'''
== Implementation Specification
(visible only on this page)

=== Message

> * Align the last closing curly bracket by adding N spaces.

Reviewer: ```suggestion
* Align the last closing curly bracket to the leftmost character present in the code by adding N spaces.
```
